op-node/p2p/cli/load_config.go

```diff
@@ -207,6 +217,30 @@ func loadDiscoveryOpts(conf *p2p.Config, ctx *cli.Context) error {
 	return nil
 }
 
+func resolveURLIP(rawurl string) (string, error) {
+	u, err := url.Parse(rawurl)
+	if err != nil {
+		return "", fmt.Errorf("parsing URL %q: %w", rawurl, err)
+	}
+	ip := net.ParseIP(u.Hostname())
+	if ip != nil {
+		return rawurl, nil
+	}
+
+	ips, err := net.LookupIP(u.Hostname())
+	if err != nil {
+		return "", fmt.Errorf("looking up IP for hostname %q: %w", u.Hostname(), err)
+	}
+	ip = ips[0]
+	// TODO: add the following check that was in geth for an IPv4 address?
+	// Ensure the IP is 4 bytes long for IPv4 addresses.
+	// if ipv4 := ip.To4(); ipv4 != nil {
+	// 	ip = ipv4
+	// }
```

Comment:
Yes, I think we should add this. But not only in this part, but also the `net.ParseIP` early return part, to match the original geth behavior. To turn IPs of enodes that do contain an IP also into a v4 IP if possible.

Also, discv5 does support ipv6 I believe, but I think the IPv4-formatted IP is preferable by the enode code.